src/plugins/intel_cpu/src/transformations/snippets/x64/op/brgemm_copy_b.cpp

```diff
@@ -107,8 +108,42 @@ std::shared_ptr<Node> intel_cpu::BrgemmCopyB::clone_with_new_inputs(const Output
                                          m_K_blk, m_N_blk);
 }
 
-size_t intel_cpu::BrgemmCopyB::get_offset_compensations() const {
+size_t BrgemmCopyB::get_offset_compensations() const {
     OPENVINO_ASSERT(is_with_compensations() && get_output_size() == 2,
                     "The offset for compensations must be in BrgemmCopyB only with compensations and 2 outputs!");
     return get_output_offset(1);
 }
+
+BrgemmCopyB::ShapeInfer::ShapeInfer(const std::shared_ptr<ov::Node>& n) {
+    const auto& brg_copyb = ov::as_type_ptr<BrgemmCopyB>(n);
+    OPENVINO_ASSERT(brg_copyb, "Got invalid node in BrgemmCopyB::ShapeInfer");
+    m_layout = snippets::lowered::PortDescriptorUtils::get_port_descriptor_ptr(n->input(0))->get_layout();
+    m_num_outs = brg_copyb->is_with_compensations() ? 2 : 1;;
+    m_N_blk = brg_copyb->get_input_element_type(0) == element::bf16 ? 32 : 64;
+    m_brgemmVNNIFactor = 4 / brg_copyb->m_src_type.size();
+}
+
+snippets::IShapeInferSnippets::Result
+BrgemmCopyB::ShapeInfer::infer(const std::vector<std::reference_wrapper<const IShapeInferSnippets::VectorDims>>& input_shapes) {
```

Comment:
Common question: Can we reuse these `shape_infer` methods in `validate_and_infer_types()` of specific ops to avoid code duplication? Or to reuse common calculations at least...

Reply:
Not sure about that, since `validate_and_infer_types` work with PartialShapes.
Probably we'll be able to reuse nGraph shapeInfer implementations one day (e.g. from [here](https://github.com/openvinotoolkit/openvino/blob/5eab1be68296f6867d5355f2a3912153059c867e/src/core/shape_inference/include/eltwise_shape_inference.hpp#L14-L33)), but:
-  We need a custom shape type that supports broadcasting, which we don't have (cause we don't really need it elsewhere)
- nGraph implementations are more universal and support cases that we don't, and perform excessive checks that we don't really need. Therefore there are still some concerns about their performance.
- Most importantly, we can reuse only a few simple cases, and the really `interesting` ones (like BrgemmCPU) still have to be reimplemented.
It's a valid point though, and we should get back to it once we start to expand our opset aggressively and enable a lot of operations with complicated shapeInfers